lib/isolator/active_record/connection_decorator.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+module Isolator
+  module ActiveRecord
+    class ConnectionDecorator < BasicObject
+      attr_reader :connection
+
+      def initialize(connection)
+        @connection = connection
+      end
+
+      def execute(sql, name = nil)
```

Comment:
Isn't watching for execute enough? 

Also, instead of monkey-patching we can rely on AS Notifications.